### PR TITLE
Show 5 most common interesting numbers, largest, and lowest numbers

### PR DESCRIPTION
**Pull request by sethtroisi**
I thought it was interesting to see a few more of the most common numbers as well as largest and lowest numbers.

Feel free to deny if you don't think this adds value.

Screenshot from 2022-05-11 17-42-43

### CODE REVIEW

File: oeis_funfacts.py [hunk 1]
```diff
@@ -101,29 +105,47 @@ def refine(results):
...
+            for i, nth in enumerate(v):
+                results[k + "_" + str(i)] = nth
+
     # JSON cannot store excessively large numbers, so we stringify all potentially-large numbers:
-    results['uninteresting_str'] = [str(e) for e in results['uninteresting']]
-    results['uninteresting_neg_str'] = [str(e) for e in results['uninteresting_neg']]
-    for k in ['largest_t', 'last_novel_t', 'lowest_t', 'most_interesting_n']:
-        results[k + '_str'] = str(results[k])
+    for k, v in list(results.items()):
+        if isinstance(v, int) and abs(v) >= 2 ** 64:
```
Comment (sethtroisi) [inline oeis_funfacts.py]: This isn't amazing because the value (e.g. `novel_t_str`) may disappear if on the next sync `novel_t <  2^64`

**Comment by BenWiederhake** [comment]
Hey, thanks for your interest! :D
I'm unable to take a close look before Monday, but I very much like the idea :)

File: oeis_funfacts.py [hunk 2]
```diff
@@ -101,29 +105,47 @@ def refine(results):
             # Should never be greater, but whatever.
             break
 
-    results['most_interesting'] = max((r.occurrence_count, n, r.first_a_number) for n, r in records.items())
     results['last_novel'] = max((r.first_a_number, n, r.occurrence_count) for n, r in records.items())
-    results['largest'] = max((n, r.first_a_number, r.occurrence_count) for n, r in records.items())
-    results['lowest'] = min((n, r.first_a_number, r.occurrence_count) for n, r in records.items())
+
+    # Grap the n "most" items from several catagories
```
Comment (BenWiederhake) [inline oeis_funfacts.py]: Grab*

File: oeis_funfacts.py (hunk 1 above)
Comment (BenWiederhake) [reply oeis_funfacts.py]: Why the restriction to large values? An interface should either be always there or never there. In this case, I'd argue that making it *always* available has no disadvantage, so let's just put it there always. (Technically it does waste a few bytes, but who cares.)

File: template.html [hunk 3]
```diff
@@ -44,16 +44,54 @@ <h2>Other uninteresting numbers</h2>
 <p class="small text-muted">(According to appearance or lack thereof in the OEIS stripped database.)</p>
 
 <h2>Interesting numbers</h2>
-<p>The most interesting number is {most_interesting_t}, with {most_interesting_n} occurrences.</p>
+<p>
+  The most interesting number is {most_interesting[0][1]}, with {most_interesting[0][0]} occurrences.<br>
+  The 2nd interesting number is {most_interesting[1][1]}, with {most_interesting[1][0]} occurrences.<br>
```
Comment (BenWiederhake) [inline template.html]: 2nd most* interesting …

**Review by BenWiederhake — COMMENTED** [review]
Love it! Do you recall the reason why you restricted the code to 2**64?

As far as I can see, it copies all entries before converting to string, so that won't be an issue.

File: oeis_funfacts.py (hunk 2 above)
Comment (sethtroisi) [reply oeis_funfacts.py]: Fixed.

File: oeis_funfacts.py (hunk 1 above)
Comment (sethtroisi) [reply oeis_funfacts.py]: Updated

File: template.html (hunk 3 above)
Comment (sethtroisi) [reply template.html]: fixed

File: oeis_funfacts.py [hunk 4]
```diff
@@ -143,7 +143,7 @@ def printablify(results):
 
     # JSON cannot store excessively large numbers, so we stringify all potentially-large numbers:
     for k, v in list(results.items()):
-        if isinstance(v, int) and abs(v) >= 2 ** 64:
+        if isinstance(v, int)
```
Comment (BenWiederhake) [inline oeis_funfacts.py:146]: That's not valid syntax. Also, why did you make the file non-executable?

File: oeis_funfacts.py [hunk 5]
```diff
@@ -49,6 +52,7 @@ def read_from(filename):
             last_modified = line[len(b'# Last Modified: '):].decode().strip()
             continue
         if line.startswith(b'#'):
+            print("Comment: ", line)
```
Comment (BenWiederhake) [inline oeis_funfacts.py:55]: What for?

File: oeis_funfacts.py [hunk 6]
```diff
@@ -101,29 +105,47 @@ def refine(results):
...
+    results['largest_t'], results['largest_a'], _ = results['largest'][0]
+    results['most_interesting_n'], results['most_interesting_t'], _ = results['most_interesting'][0]
     results['last_novel_a'], results['last_novel_t'], _ = results['last_novel']
 
+    # Unnest lists
+    for k, v in list(results.items()):
+        if isinstance(v, list):
+            print("Unnested", k)
+            for i, nth in enumerate(v):
+                results[k + "_" + str(i)] = nth
```
Comment (BenWiederhake) [inline oeis_funfacts.py:142]: This overwrites `uninteresting_1` with the second most uninteresting number, thus breaking the *main point of the website*. Come on, have you even tested it locally?

I don't see yet why you want to unnest this. It has (obviously) no advantage for the HTML version, so only whoever parses the JSON sees a difference. And if you parse JSON anyway, then the consumer really should be able to handle lists.

**Review by BenWiederhake — CHANGES_REQUESTED** [review]

**Comment by BenWiederhake** [comment]
Inactivity.

**Pull request closed without merging**